recipes-support/provisioning/files/phytec-board-config.sh

```diff
@@ -326,6 +328,18 @@ The next steps are:
             set_onboarded accountdevice
         fi
         echo "response: ${response}"
+        while true; do
+            read -p "The awsclient needs to be restarted. Restart now? (y/n)" yn
+            case $yn in
+                [Yy]* ) do_restart_awsclient; break;;
+                [Nn]* ) exit;;
+            * ) echo "Please answer yes or no.";;
+            esac
+        done
+        awsrestart="1"
+    fi
+    if [ "$awsrestart" = "0" ]; then
+        do_restart_awsclient
```

Comment:
The while loop is executed during the non-interactive part, but you add user interaction to it. I'd suggest restarting the awsclient without asking in any case, if it is necessary. This also means that the function `do_restart_awsclient` can be dissolved into the code part here because it is not used elsewhere.

Reply:
this while loop is needed since it is possible to execute an onboarding via commandline. Without this prompt the restart process would start immediately after generating the token and UID, so the user won't have the time to add the device on the platform.

Reply:
I understand. But still, we cannot ask for user input in a non-interactive program. This makes scripting `phytec-board-config` with the non-interactive options impossible. I would suggest adding an extra option that restarts the `awsclient` separately in this case. For the interactive UI we can add a part that asks for user input though.